kolibri/core/assets/test/session-nav-widget.js

```diff
@@ -0,0 +1,110 @@
+/* eslint-env mocha */
+// The following two rules are disabled so that we can use anonymous functions with mocha
+// This allows the test instance to be properly referenced with `this`
+/* eslint prefer-arrow-callback: "off", func-names: "off" */
+
+'use strict';
+
+const Vue = require('vue-test');
+const Vuex = require('vuex');
+const sessionNavWidget = require('../src/vue/session-nav-widget');
+const SessionNavWidgetComponent = Vue.extend(sessionNavWidget);
+const assert = require('assert');
+const UserKinds = require('../src/constants').UserKinds;
+
+
+describe('session-nav-widget component', function () {
+  describe('computed properties', function () {
+    beforeEach(function () {
+      this.name = 'The Best User';
+      this.username = 'and everyone knows it';
+      this.store = new Vuex.Store({
+        state: {
+          core: {
+            session: {
+              kind: [],
+              full_name: this.name,
+              username: this.username,
+            },
+          },
+        },
+      });
+      this.vm = new SessionNavWidgetComponent({ store: this.store }).$mount();
+    });
+    describe('initial', function () {
+      beforeEach(function () {
+        this.nameInitial = this.name[0].toUpperCase();
+        this.usernameInitial = this.username[0].toUpperCase();
```

Comment:
cc @rtibbles I'm not too familiar with testing best practices. However it seems intuitively 'wrong' to me to compute the expected values here rather than hard-coding them. Otherwise we need tests for the tests!